src/components/ComponentSettingsModal.jsx

```diff
@@ -0,0 +1,467 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Base64 } from 'js-base64';
+import PropTypes from 'prop-types';
+import * as _ from 'lodash';
+import { Redirect } from 'react-router-dom';
+import FormField from './FormField';
+import FormFieldFile from './FormFieldFile';
+import LoadingButton from './LoadingButton';
+import {
+  frontMatterParser,
+  concatFrontMatterMdBody,
+  enquoteString,
+  dequoteString,
+  generateResourceFileName,
+  generateCollectionPageFileName,
+  generatePageFileName,
+  retrieveCollectionAndLinkFromPermalink,
+} from '../utils';
+import elementStyles from '../styles/isomer-cms/Elements.module.scss';
+import { validatePageSettings, validateResourceSettings } from '../utils/validators';
+import DeleteWarningModal from './DeleteWarningModal';
+
+// axios settings
+axios.defaults.withCredentials = true
+
+export default class ComponentSettingsModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      permalink: '',
+      fileUrl: null,
+      date: '',
+      mdBody: null,
+      category: '',
+      sha: '',
+      allCategories: null,
+      type: '',
+      errors: {
+        title: '',
+        permalink: '',
+        fileUrl: '',
+        date: '',
+        category: '',
+        prevCategory: '',
+      },
+      canShowDeleteWarningModal: false,
+      baseApiUrl: '',
+      prevCategory: '',
+      thirdNavTitle: '',
+      redirectToNewPage: false,
+      newPageUrl: '',
+      isPost: true,
+    };
+  }
+
+  async componentDidMount() {
+    try {
+      const {
+        category, siteName, fileName, isNewFile, type
+      } = this.props;
+      if (type === "resource") {
+        const resourcesResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources`);
+        const { resources: allCategories } = resourcesResp.data;
+        this.setState({
+          category: category ? category : allCategories[0].dirName,
+          allCategories: allCategories.map((category) => category.dirName),
+        })
+      } else if (type === "page") {
+        const collectionsResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/collections`);
+        const { collections: collectionCategories } = collectionsResp.data;
+        const allCategories = [''].concat(collectionCategories)
+        this.setState({
+          category: category ? category : allCategories[0],
+          allCategories,
+        })
+      }
+
+      const baseApiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}${category ? type === "resource" ? `/resources/${category}` : `/collections/${category}` : ''}`
+      if (isNewFile) {
+        this.setState({
+          title: 'Title',
+          permalink: 'permalink',
+          mdBody: '',
+          baseApiUrl,
+          type,
+          prevCategory: category,
+        });
+        if (type === "resource") this.setState({date:'2019-01-31'})
```

Comment:
Why is the date hard-coded here?

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/ae21996696944e51e26a7e0f97f545c8cd7dd99c